### PR TITLE
Use cleanWhiteSpace to delete multiple spaces between words

### LINKED ISSUE #416: clean whitespace in form data when saving
Whenever we save form data to a model, we need to use the `.trim()` method to remove any leading/trailing whitespace.

This needs to be done for any existing fields, and we should keep it in mind when working with new fields in the future.

# Locations to Update

- [x] `Note.js`: the `text` and `source` properties in the `save()` method
- [x] `AdditionalName.js`: the `name` and `language` properties in the `save()` method
- [x] `LanguageEditor.js`
  - [x] the `name` property in the `updateProperty()` method
  - [x] the `autonym` property in the `updateAutonym()` method (change `value` on line 215 to `value.trim()`)
  - [x] the `name` property in the `handleNamesUpdate()` method

# Clean Whitespace

- [x] trim whitespace
- [x] replace multiple whitespace with just 1 whitespace

This should go in `src/utilities/cleanWhitespace.js`.

```js
function cleanWhitespace(str) {
  return str.trim().replace(/\s{2,}/gu, ` `);
}
```

### PR DESCRIPTION
**Pull request by TongJ05**
**Related Issue:**
closes #416 
**Description of Changes**
Create a separate method to clean spaces in the beginning, at the end of a word, or multiple spaces between words and use this method instead of trim() on inputs.

### CODE REVIEW

**Review by vadekh — CHANGES_REQUESTED** [review]
This will not work until the new `cleanWhitespace` function is imported into all of the other files that use it. It should be an easy change, most of the files already import other utilities so just copy that format.

**Review by dwhieb — CHANGES_REQUESTED** [review]
The way `cleanWhiteSpace` is currently written, it needs to have the string passed to it, instead of being chained on the end of the string:

❌ `string.cleanWhiteSpace()`
✅ `cleanWhiteSpace(string)`

File: src/pages/Languages/LanguageEditor/LanguageEditor.js [hunk 1]
```diff
@@ -363,7 +364,7 @@ export default class LanguageEditor extends View {
     const isValid = input.checkValidity();
     input.reportValidity();
     if (!isValid) return;
-    this.language[name] = value;
+    this.language[cleanWhitespace(name)] = value;
```
Comment (dwhieb) [inline src/pages/Languages/LanguageEditor/LanguageEditor.js]:
```suggestion
    this.language[name] = cleanWhitespace(value);
```

**Review by dwhieb — DISMISSED** [review]
Looks good! Made one minor correction (`cleanWhitespace` was being called on the wrong variable), and also added a spot I missed previously (the language name). I checked each of the fields in the Language Editor on my machine by typing in tabs or multiple spaces to see if they saved correctly, and they did.

**Pull request merged**